src/audio.cpp

```diff
@@ -54,33 +54,34 @@ void IRAM_ATTR AudioSamplerTaskEntry(void *)
 
     for (;;)
     {
-        uint64_t lastFrame = millis();
+        auto lastFrame = millis();
 
         g_Analyzer.RunSamplerPass();
         g_Analyzer.UpdatePeakData();
         g_Analyzer.DecayPeaks();
 
         // VURatio with a fadeout
 
-        static float lastVU = 0.0;
-        constexpr auto VU_DECAY_PER_SECOND = 4.0;
+        static auto lastVU = 0.0f;
+        constexpr auto VU_DECAY_PER_SECOND = 1.25f;
+
+        // Get the elapsed time since the last frame. We'll calculate this at the right spot from the first loop onwards
+        static auto frameDurationSeconds = (millis() - lastFrame) / 1000.0f;
 
         // Fade out the VU ratio... The current peak goes up at 10x the rate it comes back down,
-        // which is controlled by this constant
+        // which is controlled by VU_REACTIVITY_RATIO.
 
         if (g_Analyzer._VURatio > lastVU)
-            lastVU += (millis() - lastFrame) / 1000.0 * VU_DECAY_PER_SECOND * VU_REACTIVITY_RATIO;
+            lastVU += frameDurationSeconds * VU_DECAY_PER_SECOND * VU_REACTIVITY_RATIO;
         else
-            lastVU -= (millis() - lastFrame) / 1000.0 * VU_DECAY_PER_SECOND;
+            lastVU -= frameDurationSeconds * VU_DECAY_PER_SECOND;
 
-        lastVU = std::max(lastVU, 0.0f);
-        lastVU = std::min(lastVU, 2.0f);
-        g_Analyzer._VURatioFade = lastVU;
+        g_Analyzer._VURatioFade = std::clamp(lastVU, 0.0f, g_Analyzer._VURatio);

```

Comment:
This is the actual fix for the problem mentioned in the PR description. The VU ratio wasn't capped to the value of `g_Analyzer._VURatio`, which it has to be.